Unix/tests/pal/test_pal.cpp

```diff
@@ -35,7 +35,18 @@
 #include <pal/file.h>
 #include <pal/hashmap.h>
 #include <pal/format.h>
+
+#if !defined(linux) && !defined(macos) && defined(CONFIG_ENABLE_WCHAR)
```

Comment:
I'd prefer that you called out where this did NOT work rather than do negative comparisons for where it does work. Something like:

```
#if ( defined(aix) || defined(hpux) || defined(sun) ) && defined(CONFIG_ENABLE_WCHAR)
```

That way, you're specifically calling out where it does NOT work, and it's easier to fix a specific (non-working) platform if it gets fixed with newer compilers.